dask/bytes/local.py

```diff
@@ -87,10 +116,25 @@ def open_files(path):
             for _path in filepaths]
 
 
+def write_files(paths, mode='wb'):
+    """ Open many files for writing.  Return delayed objects.
+
+    See Also
+    --------
+    dask.bytes.core.write_files: User function
+    """
+    myopen = delayed(open)
+    files = [myopen(_path, mode=mode,
+                   dask_key_name='write-%s' % tokenize(_path, time.time()))
+            for _path in paths]
+    return files
```

Comment:
This comment gets to the heart of what I was struggling with. The current scheme opens files when evaluated (not before) and another delayed function would write to it. This is how I interpreted what goes on in open_files and in the previous to_csv.